code/modules/projectiles/guns/projectile/constructable/swords.dm

```diff
@@ -24,6 +24,7 @@
 		return
 	to_chat(user, "You remove the hypospray from \the [src].")
 	var/obj/item/weapon/reagent_containers/hypospray/H = new (get_turf(user.loc))
+	H.reagents.remove_reagent("doctorsdelight", 30)
```

Comment:
Seems to me like the proper fix would be adding the original hypospray as a var in the Venom sword (with loc set inside the sword itself) so it returns the same hypospray used to construct it, with the same chemicals and fingerprints